artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumProjectIntegration/Features/Projects/GetUserProjectsService.cs

```diff
@@ -0,0 +1,65 @@
+using Artificial.Scrum.Master.ScrumProjectIntegration.Exceptions;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ApiTokens;
+using System.Net;
+using System.Text.Json;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ScrumServiceHttpClient;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Utilities;
+
+namespace Artificial.Scrum.Master.ScrumProjectIntegration.Features.Projects
+{
+    internal interface IGetUserProjectsService
+    {
+        Task<IEnumerable<GetUserProjectsResponse>> Handle(string userId);
+    }
+
+    internal class GetUserProjectsService : IGetUserProjectsService
+    {
+        private readonly IUserTokensRepository _userTokensRepository;
+        private readonly IProjectHttpClientWrapper _projectHttpClientWrapper;
+        private readonly IJwtDecoder _jwtDecoder;
+
+        public GetUserProjectsService(IUserTokensRepository userTokensRepository,
+            IProjectHttpClientWrapper projectHttpClientWrapper, IJwtDecoder jwtDecoder)
+        {
+            _userTokensRepository = userTokensRepository;
+            _projectHttpClientWrapper = projectHttpClientWrapper;
+            _jwtDecoder = jwtDecoder;
+        }
+
+        public async Task<IEnumerable<GetUserProjectsResponse>> Handle(string userId)
+        {
+            var userTokens = await _userTokensRepository.GetUserAccessTokens(userId);
+            if (userTokens is null)
+            {
+                throw new ProjectResourceNotFoundException($"Credentials of user:{userId} not found");
+            }
+
+            var projectRequestResult = await _projectHttpClientWrapper.GetHttpRequest(
+                userId,
+                $"projects?member={_jwtDecoder.GetClaim(userTokens.Value.AccessToken, "user_id")}"
+            );
+
+            if (projectRequestResult.StatusCode != HttpStatusCode.OK)
```

Comment:
a te request failed to możesz też przenieść do środka repozytorium(klientWrappera) żeby on zwracał gotowe obiekty jak w generyce<T>